src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.dev.cinema.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+import com.dev.cinema.dao.MovieSessionDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.MovieSession;
+import com.dev.cinema.util.HibernateUtil;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    private static final Logger LOGGER = Logger.getLogger(MovieSessionDaoImpl.class);
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movieSession);
+            transaction.commit();
+            movieSession.setId(id);
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction == null) {
+                transaction.rollback();
+            }
+            LOGGER.error("Cannot add movie session to database", e);
+            throw new RuntimeException();
```

Comment:
suggestion: throw new RuntimeException("Cannot add movie session to database", e);

Reply:
done